components/schema-migrator/migrations/director/202102031458_extend_application_resources.up.sql

```diff
@@ -0,0 +1,102 @@
+BEGIN;
+
+ALTER TABLE applications
+    ADD COLUMN ready bool NOT NULL DEFAULT TRUE,
+    ADD COLUMN created_at timestamp NOT NULL,
+    ADD COLUMN updated_at timestamp NOT NULL,
+    ADD COLUMN deleted_at timestamp NOT NULL,
+    ADD COLUMN error jsonb;
+
+ALTER TABLE bundles
+    ADD COLUMN ready bool NOT NULL DEFAULT TRUE,
+    ADD COLUMN created_at timestamp NOT NULL,
+    ADD COLUMN updated_at timestamp NOT NULL,
+    ADD COLUMN deleted_at timestamp NOT NULL,
+    ADD COLUMN error jsonb;
+
+ALTER TABLE api_definitions
+    ADD COLUMN ready bool NOT NULL DEFAULT TRUE,
+    ADD COLUMN created_at timestamp NOT NULL,
+    ADD COLUMN updated_at timestamp NOT NULL,
+    ADD COLUMN deleted_at timestamp NOT NULL,
+    ADD COLUMN error jsonb;
+
+ALTER TABLE event_api_definitions
+    ADD COLUMN ready bool NOT NULL DEFAULT TRUE,
+    ADD COLUMN created_at timestamp NOT NULL,
+    ADD COLUMN updated_at timestamp NOT NULL,
+    ADD COLUMN deleted_at timestamp NOT NULL,
+    ADD COLUMN error jsonb;
+
+ALTER TABLE documents
+    ADD COLUMN ready bool NOT NULL DEFAULT TRUE,
+    ADD COLUMN created_at timestamp NOT NULL,
+    ADD COLUMN updated_at timestamp NOT NULL,
+    ADD COLUMN deleted_at timestamp NOT NULL,
+    ADD COLUMN error jsonb;
+
+ALTER TABLE applications
+    ADD CONSTRAINT application_id_ready_unique UNIQUE (id, ready);
+
+ALTER TABLE bundles
+    ADD CONSTRAINT bundle_id_ready_unique UNIQUE (id, ready);
+
+ALTER TABLE api_definitions
+    ADD CONSTRAINT api_id_ready_unique UNIQUE (id, ready);
+
+ALTER TABLE event_api_definitions
+    ADD CONSTRAINT event_api_id_ready_unique UNIQUE (id, ready);
+
+ALTER TABLE documents
+    ADD CONSTRAINT document_id_ready_unique UNIQUE (id, ready);
+
+ALTER TABLE bundles
+    ADD CONSTRAINT bundles_applications_ready_fk
+        FOREIGN KEY (app_id, ready) REFERENCES applications (id, ready) ON UPDATE CASCADE;
```

Comment:
havent looked at the tests yet but do we have tests for that one ? Basically marking an as ready marks  apis, events as ready as well.

Reply:
We've only tested this manually. Maybe it won't be trivial to make a clean test for this as this would require making an async request, which we still don't have a full implementation of, we'll see if something can be done nevertheless

Reply:
These tests will be part of the operations controller e2e tests.